realm/realm-library/src/androidTest/java/io/realm/RealmObjectTests.java

```diff
@@ -1558,19 +1558,39 @@ public void setter_nullValueInNullableField() {
 
     @Test
     @RunTestInLooperThread
-    public void addChangeListener_throwOnAddingNullListener() {
+    public void addChangeListener_throwOnAddingNullListenerFromLooperThread() {
         final Realm realm = looperThread.realm;
         Dog dog = createManagedDogObjectFromRealmInstance(realm);
 
         try {
-            dog.addChangeListener((RealmChangeListener) null);
-            fail("Failed on adding null change listener.");
+            dog.addChangeListener(null);
```

Comment:
To minimize the chance of unintentional pass of this test case(i.e. the case of `createManagedDogObjectFromRealmInstance ()`'s throwing `IllegalArgumentException`), I would keep this(and update other inlined codes).